lua/ssr/utils.lua

```diff
@@ -55,13 +54,18 @@ end
 
 -- Get smallest node for the range.
 ---@param buf buffer
+---@param lang string
 ---@param start_row number
 ---@param start_col number
 ---@param end_row number
 ---@param end_col number
----@return TSNode
-function M.node_for_range(buf, start_row, start_col, end_row, end_col)
-  return parsers.get_parser(buf):parse()[1]:root():named_descendant_for_range(start_row, start_col, end_row, end_col)
+---@return TSNode?
+function M.node_for_range(buf, lang, start_row, start_col, end_row, end_col)
+  local has_parser, parser = pcall(vim.treesitter.get_parser, buf, lang)
+  if not has_parser then
```

Comment:
The parser must present since we checked it at the beginning, so this should be not needed.

Reply:
Sorry for this, this should be needed.